src/query-builder/SelectQueryBuilder.ts

```diff
@@ -3750,8 +3769,17 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
 
     /**
      * Loads raw results from the database.
+     * @param {Date} [timestamp] A timestamp to get datasets at a specific time. If undefined use a date in the future to get the current datasets.
      */
-    protected async loadRawResults(queryRunner: QueryRunner) {
+    protected async loadRawResults(
+        queryRunner: QueryRunner,
+        timestamp = dayjs().add(10, "year").toDate(),
+    ) {
+        this.setParameter(
+            "timestamp",
```

Comment:
shadow parameter. User may easily override / get lost of his own parameter with the same name.

Reply:
I change the parameter name to `temporalTableValidAt`.